### PR TITLE
Fix edit user

### PR DESCRIPTION
**Pull request by soupette**

### CODE REVIEW

File: packages/strapi-plugin-content-manager/config/policies/routing.js [hunk 1]
```diff
@@ -8,7 +8,40 @@ module.exports = async (ctx, next) => {
 
     if (controller && action) {
       // Redirect to specific controller.
-      return await strapi.plugins[source].controllers[controller.toLowerCase()][action](ctx);
+      if (ctx.request.body.hasOwnProperty('fields') && ctx.request.body.hasOwnProperty('files')) {
+        const {files, fields} = _.cloneDeep(ctx.request.body);
+
+        const fileAttributes = _.get(strapi.plugins, [source, 'models', ctx.params.model, 'associations'], []).filter(association => {
+          return association[association.type] === 'file' && association.via === 'related';
+        }).reduce((acc, association) => {
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-manager/config/policies/routing.js]:
```js
.map(association => association.alias);
```

?

File: packages/strapi-plugin-content-manager/config/policies/routing.js [hunk 2]
```diff
@@ -8,7 +8,40 @@ module.exports = async (ctx, next) => {
...
+          acc.push(association.alias);
+          return acc;
+        }, []);
+
+        ctx.request.body = _.omit(fields, fileAttributes);
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-manager/config/policies/routing.js]: Not sure to understand why you're omitting file relationships... What's happening if I'm removing a file from a field which can take multiple files. I think we need these fields to be able to remove the relationships.

File: packages/strapi-plugin-content-manager/config/policies/routing.js [hunk 3]
```diff
@@ -8,7 +8,40 @@ module.exports = async (ctx, next) => {
...
+
+        await strapi.plugins[source].controllers[controller.toLowerCase()][action](ctx);
+        const resBody = _.cloneDeep(ctx.body);
+
+        Object.keys(files).map(async field => {
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-manager/config/policies/routing.js]: Promise.all() to upload asynchronously the files of each field?

File: packages/strapi-plugin-content-manager/config/policies/routing.js [hunk 4]
```diff
@@ -8,7 +8,40 @@ module.exports = async (ctx, next) => {
...
+              ref: ctx.params.model,
+              source,
+              field
+            }
+          };
+
+          await strapi.plugins.upload.controllers.upload.upload(ctx);
+        });
+        return ctx.send(resBody);
+      } else {
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-content-manager/config/policies/routing.js]: Unnecessary condition `else`?

File: packages/strapi-plugin-content-manager/config/policies/routing.js (hunk 2 above)
Comment (lauriejim) [reply packages/strapi-plugin-content-manager/config/policies/routing.js]: Exact we can't delete files when it's multiple file ! But actually if I send relationships, it's break the relation between files. I will investigate.

**Pull request merged**